### PR TITLE
170869185 user should be able to sign up

### PR DESCRIPTION
**Pull request by izzett222**
#### What does this PR do?
adds the signup endpoint
#### Description of Task to be completed?
- added signup test
- added the signup controller
- added signup request data validator middleware
#### How should this be manually tested?
- clone this repo
- run npm install
- create the .env file
- npm run test
#### What are the relevant pivotal tracker stories?
#170869185
#### Screenshots (if appropriate)
Screen Shot 2020-01-27 at 18 31 08

### CODE REVIEW

**Comment by blaise82** [comment]
Good job

**Review by RusimbiPatrick — COMMENTED** [review]
looks good

File: server/controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        fullName: fullName.trim(),
+        email: email.trim(),
+        password: await bcrypt.hash(password, 10),
+        isAdmin: false,
+      });
+      const token = jwt.sign({
+        email: newUser.email,
+        isAdmin: newUser.isAdmin,
+
+      }, process.env.JWT_KEY);
```
Comment (AbdoulNuru) [inline server/controllers/user.js]: Wouldn't it be better if we use helpers for these functions and just call them in the controllers?  That would increase our maintainability I guess.

Comment (blaise82) [reply server/controllers/user.js]: Yeah it would help us safe time

**Pull request merged**